File: app/src/main/java/com/isp/app/ui/schedule/ScheduleAdapter.java
package com.isp.app.ui.schedule;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.isp.app.R;
import java.util.List;

public class ScheduleAdapter extends RecyclerView.Adapter<ScheduleAdapter.ScheduleViewHolder> {

    private final List<ScheduleItem> scheduleItems;

    public ScheduleAdapter(List<ScheduleItem> scheduleItems) {
        this.scheduleItems = scheduleItems;
    }

    @NonNull
    @Override
    public ScheduleViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_schedule, parent, false);
        return new ScheduleViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ScheduleViewHolder holder, int position) {
        ScheduleItem item = scheduleItems.get(position);
        holder.courseNameTextView.setText(item.getCourseName());
        holder.scheduleTimeTextView.setText(item.getScheduleTime());
    }

    @Override
    public int getItemCount() {
        return scheduleItems.size();
    }

    static class ScheduleViewHolder extends RecyclerView.ViewHolder {
        TextView courseNameTextView;
        TextView scheduleTimeTextView;

        ScheduleViewHolder(View itemView) {
            super(itemView);
            courseNameTextView = itemView.findViewById(R.id.courseName);
            scheduleTimeTextView = itemView.findViewById(R.id.scheduleTime);
        }
    }
} 

File: app/src/main/java/com/college/app/ui/schedule/adapter/ScheduleAdapter.kt
package com.college.app.ui.schedule.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.college.app.data.models.ClassSchedule
import com.college.app.databinding.ItemClassScheduleBinding

class ScheduleAdapter : ListAdapter<ClassSchedule, ScheduleAdapter.ScheduleViewHolder>(ScheduleDiffCallback()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ScheduleViewHolder {
        val binding = ItemClassScheduleBinding.inflate(
            LayoutInflater.from(parent.context),
            parent,
            false
        )
        return ScheduleViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ScheduleViewHolder, position: Int) {
        holder.bind(getItem(position))
    }

    class ScheduleViewHolder(
        private val binding: ItemClassScheduleBinding
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(classSchedule: ClassSchedule) {
            binding.apply {
                courseName.text = classSchedule.courseName
                courseTime.text = "${classSchedule.startTime} - ${classSchedule.endTime}"
                courseRoom.text = "Room: ${classSchedule.room}"
                courseType.text = classSchedule.type.capitalize()
                courseDay.text = classSchedule.day
            }
        }
    }

    private class ScheduleDiffCallback : DiffUtil.ItemCallback<ClassSchedule>() {
        override fun areItemsTheSame(oldItem: ClassSchedule, newItem: ClassSchedule): Boolean {
            return oldItem.courseId == newItem.courseId && 
                   oldItem.startTime == newItem.startTime &&
                   oldItem.day == newItem.day
        }

        override fun areContentsTheSame(oldItem: ClassSchedule, newItem: ClassSchedule): Boolean {
            return oldItem == newItem
        }
    }
} 